source/tests/system/nidcpower_session_tests.cpp

```diff
@@ -151,27 +154,36 @@ TEST_F(NiDCPowerSessionTest, InvalidSession_CloseSession_ReturnsInvalidSessionEr
   nidevice_grpc::Session session;
   session.set_id(0UL);
 
-  ::grpc::ClientContext context;
-  dcpower::CloseRequest request;
-  request.mutable_vi()->set_id(session.id());
-  dcpower::CloseResponse response;
-  ::grpc::Status status = GetStub()->Close(&context, request, &response);
-
-  EXPECT_EQ(::grpc::StatusCode::UNKNOWN, status.error_code());
-  auto error = json::parse(status.error_message());
-  EXPECT_EQ(kInvalidDCPowerSession, error.value("code", 0));
-  EXPECT_THAT(error.value("message", "").c_str(), HasSubstr(kInvalidDCPowerSessionMessage));
+  try {
+    ::grpc::ClientContext context;
+    dcpower::CloseRequest request;
+    request.mutable_vi()->set_id(session.id());
+    dcpower::CloseResponse response;
+    auto status = GetStub()->Close(&context, request, &response);
+    nidevice_grpc::experimental::client::raise_if_error(status, context);
+    EXPECT_FALSE(true);
+  }
+  catch (const nidevice_grpc::experimental::client::grpc_driver_error& ex) {
+    EXPECT_EQ(::grpc::StatusCode::UNKNOWN, ex.StatusCode());
+    const auto& error = ex.Trailers().find("ni-error")->second;
+    EXPECT_EQ(kInvalidDCPowerSession, std::stoi(error));
+    EXPECT_THAT(ex.what(), HasSubstr(kInvalidDCPowerSessionMessage));
+  }
 }
 
-TEST_F(NiDCPowerSessionTest, InitWithErrorFromDriver_ReturnsUserErrorMessage)
+TEST_F(NiDCPowerSessionTest, InitWithErrorFromDriver_ReturnsDriverErrorWithUserErrorMessage)
```

Comment:
Note to reviewers: Several of these classes had 99%-identical tests for the 2 conditions. I consolidated them into a single test to make it easier on me.